### PR TITLE
Adds a new spell to Heretic starting loadout, "Shadow Cloak", which conceals their identity.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Heretics now start with a new spell, "Shadow Cloak".
   - Requires focus to use.
   - Hulks cannot use this spell
   - Shadow Cloak completely hides the heretic's identity for three minutes, or until cancelled. 
      - All speech shows up as "Unknown" and examines are blocked, same as the Syndicate Sneaksuit
      - This also conceals your sprite, hiding even held items
      - However, you are enclosed in a purple smoke, making it fairly obvious you are up to no good
   - Shadow Cloak increases move speed slightly, but drastically decreases action speed.
   - Taking over (currently: 25) damage while cloaked has a chance to reveal you with every further hit, causing negative effects. 

https://user-images.githubusercontent.com/51863163/196604255-ca4e3297-d9de-4948-b168-07437c19a87f.mp4

Potential todo ideas:
- Add other knowledges that buff Shadow Cloak / alleviates debuffs

## Why It's Good For The Game

An issue with Heretic's currently is that, despite having enforced memory loss, actually keeping your identity a secret is difficult.

Enter Shadow Cloak. You spawn with the spell, so all Heretics are able to make use of it to stay secretive. 
This spell is purely designed for heretics undertaking assassinations to stay a mystery. While it does confer a speed boost, using it in combat is not advised, as it will cause adverse effects if damage is sustained. 

## Changelog

:cl: Melbert
add: Heretics spawn with a new spell, Shadow Cloak. Using it will hide your identity for 3 minutes or until cancelled, but is fairly obvious otherwise, as it shrouds you in purple smoke. 
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
does stamina damage contribute towards decloaking?

why's the actionspeed modifier here?

**Comment by Palpetinus** [comment]
I like the spell itself, but decreasing action speed seems a little odd considering heretics already have one of the longest actions needed for saccing targets, drawing the actual rune. I think I speak for most heretics when I say that I'd much, MUCH rather decloak to draw the rune and risk getting revealed, than taking even longer with that dang rune.

**Comment by private-tristan** [comment]
I like this, but 2 things stand out
the cooldown appears to be 6 seconds, seems like heretics can, after killing their target, simply uncloak, begin drawing the rune, and then cloak again to skip the action speed debuff 

and if stamina damage will count for decloak, won't it just mean that a heretic effectively has to turn it off if they want to fight a head of staff (one of their targets) or a member of security (although I can see how this can be intentional)

otherwise i really like this

File: code/modules/antagonists/heretic/magic/shadow_cloak.dm [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	return TRUE
+
+/datum/status_effect/shadow_cloak/on_remove()
+	// Remove image
+	owner.remove_alt_appearance(id)
+	QDEL_NULL(cloak_image)
+	// Remove traits and modifiers
+	REMOVE_TRAIT(owner, TRAIT_UNKNOWN, id)
+	REMOVE_TRAIT(owner, TRAIT_SILENT_FOOTSTEPS, id)
+	owner.name = owner.get_visible_name() // This is done every life tick, we'll just do it here to make sure it's immediate
```
[the remark this answers is not in the document]
Comment (necromanceranne) [reply code/modules/antagonists/heretic/magic/shadow_cloak.dm:163]: Remove this NOW

**Pull request merged**